packages/bp/src/core/bots/bot-service.ts

```diff
@@ -326,9 +325,14 @@ export class BotService {
       } else {
         this.logger.forBot(botId).info(`Import of bot ${botId} was denied by hook validation`)
       }
+    } catch (error) {
+      const errorMessage = error && error instanceof Error ? error.message : JSON.stringify(error)
+      this.logger.error(`Error in ${this.importBot.name} function: ${errorMessage}`)
```

Comment:
In which case does it returns an error which has to be stringified ? I never heard about that

```js
catch(error){
  if(error instanceof Error){
    this.logger.attachError(error).error('Error while importing bot')
  }
}
```

Reply:
The following screenshots ,[this](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDCCuEAuB7AtgUQE5ZV6ApgB5IEB2AJjNrvgN4BQ0z0wKZyW8wqWAFAEo6EeAAcC-AQF8GMhmw5JCOPNAC80MgQDucRKkwrJDeewgoQBAHQgUAcz4Ta0AJaKwZYARQAzaDTwBIA) and a warning about using too many type checks on another PR are the reasons why I've crafted the code the way I did. The ternary operator is merely to make sure that we have a fallback (in type string) in case the thrown error is not an instance of error.
![Screenshot from 2021-09-28 15-11-34](https://user-images.githubusercontent.com/30733005/135151344-e70cefe1-1e6d-4b66-8157-58e361475cf2.png)
![Screenshot from 2021-09-28 15-11-56](https://user-images.githubusercontent.com/30733005/135151346-e861db36-e5ce-4763-ad66-2093928c8e1c.png)

Reply:
The newest version of typescript used on VS Code now considers error as unknown instead of any, so that is why you have those warnings. Whenever possible, use `attachError` on the logger to print easily the stack trace, instead of a manually crafted message including the error message